politician_analysis/src/scraper.py

```diff
@@ -0,0 +1,126 @@
+import json
+import time
+import requests
+
+
+YOUTUBE_URL = 'https://www.youtube.com/watch?v={youtube_id}'
+USER_AGENT = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36'
+SORT_BY_POPULAR = 0
+SORT_BY_RECENT = 1
+
+def find_value(html, key, num_chars=2, separator='"'):
+    pos_begin = html.find(key) + len(key) + num_chars
+    pos_end = html.find(separator, pos_begin)
+    return html[pos_begin: pos_end]
+
+
+def ajax_request(session, endpoint, version, api_key, retries=5, sleep=20):
+    
+    component = endpoint['commandMetadata']['webCommandMetadata']['apiUrl']
+    
+    url = f'https://www.youtube.com{component}'
+    data = {'context': {'client': {'userAgent': USER_AGENT, 'clientName': 'WEB', 'clientVersion': version},
+                        'clickTracking': {'clickTrackingParams': endpoint['clickTrackingParams']}},
+            "continuation": endpoint['continuationCommand']['token']}
+
+    for _ in range(retries):
+        response = session.post(url, params={'key': api_key}, json=data)
+        if response.status_code == 200:
+            return response.json()
+        elif response.status_code in [403, 413]:
+            return {}
+        else:
+            time.sleep(sleep)
+
+def search_dict(partial, search_key):
+    stack = [partial]
+    while stack:
+        current_item = stack.pop()
+        if isinstance(current_item, dict):
+            for key, value in current_item.items():
+                if key == search_key:
+                    yield value
+                else:
+                    stack.append(value)
+        elif isinstance(current_item, list):
+            for value in current_item:
+                stack.append(value)
```

Comment:
It looks like no condition judgment, can we directly use stack += current_item ?